substrate/frame/referenda/src/migration.rs

```diff
@@ -161,6 +178,114 @@ pub mod v1 {
 	}
 }
 
+/// Migration for when changing the block number provider.
+///
+/// This migration is not guarded
+pub mod switch_block_number_provider {
+	use super::*;
+
+	/// The log target.
+	const TARGET: &'static str = "runtime::referenda::migration::change_block_number_provider";
+	/// Convert from one to another block number provider/type.
+	pub trait BlockNumberConversion<Old, New> {
+		/// Convert the `old` block number type to the new block number type.
+		///
+		/// Any changes in the rate of blocks need to be taken into account.
+		fn convert_block_number(block_number: Old) -> New;
+	}
+
+	/// Transforms SystemBlockNumberFor<T> to BlockNumberFor<T,I>
+	pub struct MigrateV1ToV2<BlockConverter, T, I = ()>(
```

Comment:
name seems wrong, this doesn't set the storage version to 2.

```suggestion
	pub struct MigrateBlockNumberProvider<BlockConverter, T, I = ()>(
```